src/mlpack/methods/bayesian_linear_regression/bayesian_linear_regression.hpp

```diff
@@ -0,0 +1,284 @@
+/**
+ * @file bayesian_linear_regression.hpp
+ * @author Clement Mercier
+ *
+ * Definition of the BayesianRidge class, which performs the
+ * bayesian linear regression. According to the armadillo standards,
+ * all the functions consider data in column-major format.
+**/
+
+#ifndef MLPACK_METHODS_BAYESIAN_LINEAR_REGRESSION_HPP
+#define MLPACK_METHODS_BAYESIAN_LINEAR_REGRESSION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace regression {
+
+/**
+ * A Bayesian approach to the maximum likelihood estimation of the parameters 
+ * \f$ \omega \f$ of the linear regression model. The Complexity is governed by 
+ * the addition of a gaussian isotropic prior of precision \f$ \alpha \f$ over 
+ * \f$ \omega \f$:  
+ *
+ * \f[
+ * p(\omega|\alpha) = \mathcal{N}(\omega|0, \alpha^{-1}I)
+ * \f]
+ * 
+ * The optimization procedure calculates the posterior distribution of 
+ * \f$ \omega \f$ knowing the data by maximizing an approximation of the log 
+ * marginal likelihood derived from a type II maximum likelihood approximation. 
+ * The determination of \f$ alpha \f$ and of the noise precision \f$ beta \f$
+ * is part of the optimization process, leading to an automatic determination of 
+ * w. The model being entirely based on probabilty distributions, uncertainties 
+ * are available and easly computed for both the parameters and the predictions.
+ *
+ * The advantage over linear regression and ridge regression is that the 
+ * regularization is determined from all the training data alone without any
+ * require to an hold out method. 
+ *
+ * The code below is an implementation of the maximization of the evidence 
+ * function described in the section 3.5.2 of the C.Bishop book, Pattern
+ * Recognition and Machine Learning.  
+ *
+ * @code
+ * @article{MacKay91bayesianinterpolation,
+ *   author = {David J.C. MacKay},
+ *   title = {Bayesian Interpolation},
+ *   journal = {NEURAL COMPUTATION},
+ *   year = {1991},
+ *   volume = {4},
+ *   pages = {415--447}
+ * }
+ * @endcode
+ *
+ * @book{Bishop:2006:PRM:1162264,
+ *   author = {Bishop, Christopher M.},
+ *   title = {Pattern Recognition and Machine Learning (Information Science 
+ *            and Statistics)},
+ *   chapter = {3}
+ *   year = {2006},
+ *   isbn = {0387310738},
+ *   publisher = {Springer-Verlag},
+ *   address = {Berlin, Heidelberg},
+ * } 
+ * @encode
```

Comment:
```suggestion
 * @code
 * @book{Bishop:2006:PRM:1162264,
 *   author = {Bishop, Christopher M.},
 *   title = {Pattern Recognition and Machine Learning (Information Science 
 *            and Statistics)},
 *   chapter = {3}
 *   year = {2006},
 *   isbn = {0387310738},
 *   publisher = {Springer-Verlag},
 *   address = {Berlin, Heidelberg},
 * } 
 * @endcode
```

I think that this will fix the build issue for the LaTeX documentation: http://ci.mlpack.org/job/pull-requests%20doxygen%20check/5136/console